scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Record.java

```diff
@@ -839,21 +865,17 @@ public void setSequenceNumber(long sequenceNumber) throws GeneralSecurityExcepti
 	 * @return the fragment length excluding record headers
 	 */
 	public int getFragmentLength() {
-		return fragmentLength;
+		final byte[] bytes = fragmentBytes;
+		return bytes == null ? 0 : bytes.length;
```

Comment:
When the length is unknown, should we return -1 ? 
Or something which express clearly this could not be a real fragment length ? 

(and add this to this javadoc?)

Reply:
I've updated the constructors to ensure, that the fragmentBytes are available.